exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonReader.java

```diff
@@ -91,9 +91,7 @@ public void testFieldSelectionBug() throws Exception {
           .baselineColumns("col_1", "col_2")
           .baselineValues(
               mapOf(),
-              mapOf(
-                  "inner_1", listOf(),
-                  "inner_3", mapOf()))
+              mapOf())
```

Comment:
The test file used here is not sufficient to fully test the code change. Perhaps create an additional test, using a new test file, that tests all the variations suggested in the comment below and in the JIRA entry.